packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -655,7 +655,7 @@ Future<Map<String, dynamic>> _deviceToMap(Device device) async {
     'id': device.id,
     'name': device.name,
     'platform': getNameForTargetPlatform(await device.targetPlatform),
-    'emulator': device.isLocalEmulator
+    'emulator': await device.isLocalEmulator,
```

Comment:
why are we using a map here instead of a structured class? it would have caught this problem.